### ISSUE TITLE
[docs] Add doctests for parameter schedulers

### ISSUE THREAD

**Issue by ydcjeff** [pytorch/ignite#2266]
### 📚 Documentation

Following #2230 , I listed the parameter schedulers places (suggested by @sdesrozis) where doctests can be added.

- [x] StateParamScheduler (Abstract class, no test needed)
- [x] ConcatScheduler (#2384) 
- [x] CosineAnnealingScheduler (#2384)
- [x] CyclicalScheduler (Abstract class, no test needed)
- [x] LRScheduler (#2389)
- [x] LinearCyclicalScheduler (#2327)
- [x] ParamGroupScheduler (#2389)
- [x] ParamScheduler (Abstract class, no test needed)
- [x] PiecewiseLinear (#2290)
- [x] create_lr_scheduler_with_warmup (#2389)
- [x] LambdaStateScheduler (#2402)
- [x] PiecewiseLinearStateScheduler (#2400)
- [x] ExpStateScheduler (#2391)
- [x] StepStateScheduler (#2394)
- [x] MultiStepStateScheduler (#2399)

This is a great issue for first comers and those who want to learn more about Sphinx. This issue is the community effort.
If you want to contribute, please comment on this issue with the metric you want to add doctest (one by one).

Example on how to add doctest, see #2241 

Thanks!
